core/src/main/java/org/apache/calcite/rel/rules/ReduceExpressionsRule.java

```diff
@@ -1173,6 +1180,14 @@ private void analyzeCall(RexCall call, Constancy callConstancy) {
     /** Sets {@link #matchNullability()}. */
     Config withMatchNullability(boolean matchNullability);
 
+    /** Whether to treat dynamic functions as non-constants. */
+    @ImmutableBeans.Property
+    @ImmutableBeans.BooleanDefault(true)
+    boolean treatDynamicCallsAsNonConstant();
```

Comment:
What if the naming was `treatDynamicCallsAsConstants` with the default value of `false`?
It looks like `not` adds complexity like "oh, what else could it be besides constants?"

Reply:
Agreed. I prefer not using negative naming too. I've renamed it to `treatDynamicCallsAsConstants` as suggested.